main.go

```diff
@@ -306,7 +326,31 @@ func protoToSamples(req *prompb.WriteRequest) model.Samples {
 
 func sendSamples(w writer, samples model.Samples) error {
 	begin := time.Now()
-	err := w.Write(samples)
+	shouldWrite := true
+	if election != nil {
+		isLeader, err := election.IsLeader()
```

Comment:
`isLeader` seems redundant. Can't you just use `shouldWrite` here?

Reply:
Sure, again I did it only to make it easier understanding what's going on. I am not strongly opinionated on this one, so we can remove that extra var?

Reply:
Was actually confusing to me to see both, since I first tried to figure out the difference between the two variables, given their names. But this might be a personal thing.

Reply:
I'm with Erik, I having both is more confusing.